### PR TITLE
Backport/1069 move more utils

### PR DESCRIPTION
**Pull request by xavierh**
Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/Field.java [hunk 1]
```diff
@@ -1,13 +1,14 @@
 package com.minecolonies.coremod.entity.ai.citizen.farmer;
 
 import com.minecolonies.api.colony.permissions.Action;
+import com.minecolonies.api.util.constant.Constants;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/Field.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Constants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/PathNavigate.java [hunk 2]
```diff
@@ -3,7 +3,8 @@
 import com.minecolonies.api.util.BlockPosUtil;
 import com.minecolonies.coremod.entity.EntityCitizen;
 import com.minecolonies.api.util.BlockUtils;
-import com.minecolonies.coremod.util.Log;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.coremod.entity.EntityCitizen;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/PathNavigate.java]: MINOR Remove this duplicated import. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java [hunk 3]
```diff
@@ -1,7 +1,9 @@
 package com.minecolonies.coremod.colony.buildings;
 
-import com.minecolonies.api.util.constants.ToolType;
+import com.minecolonies.api.util.constant.ToolType;
 import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.LanguageHandler;
+import com.minecolonies.api.util.Utils;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java]: MINOR Remove this unused import 'com.minecolonies.api.util.Utils'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/tileentities/ScarecrowTileEntity.java [hunk 4]
```diff
@@ -1,10 +1,11 @@
 package com.minecolonies.coremod.tileentities;
 
 import com.minecolonies.api.util.EntityUtils;
+import com.minecolonies.api.util.LanguageHandler;
+import com.minecolonies.api.util.constant.Constants;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/tileentities/ScarecrowTileEntity.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Constants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/WorkManager.java [hunk 5]
```diff
@@ -1,7 +1,7 @@
 package com.minecolonies.coremod.colony;
 
+import com.minecolonies.api.util.Log;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/WorkManager.java]: MINOR Remove this unused import 'com.minecolonies.api.util.Log'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/network/messages/OpenInventoryMessage.java [hunk 6]
```diff
@@ -1,13 +1,14 @@
 package com.minecolonies.coremod.network.messages;
 
 import com.minecolonies.api.colony.permissions.Action;
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.coremod.MineColonies;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/network/messages/OpenInventoryMessage.java]: MINOR Remove this unused import 'com.minecolonies.coremod.MineColonies'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java [hunk 7]
```diff
@@ -1,8 +1,9 @@
 package com.minecolonies.coremod.colony.buildings;
 
 import com.minecolonies.api.util.BlockPosUtil;
-import com.minecolonies.api.util.constants.ToolType;
+import com.minecolonies.api.util.constant.ToolType;
 import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.Utils;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java]: MINOR Remove this unused import 'com.minecolonies.api.util.Utils'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 294 issues
* ![MAJOR][MAJOR] 173 major
* ![MINOR][MINOR] 70 minor
* ![INFO][INFO] 51 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutMiner.java#L59](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/client/gui/WindowHutMiner.java#L59): Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L859](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L859): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L956](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L956): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.269 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L667](https://github.com/Minecolonies/minecolonies/blob/3cbecdf8b8ddb4a1a4eb7f59ae5a9790bb5bfcb1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L667): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 287 issues
* ![MAJOR][MAJOR] 173 major
* ![MINOR][MINOR] 63 minor
* ![INFO][INFO] 51 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutMiner.java#L59](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/client/gui/WindowHutMiner.java#L59): Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L859](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L859): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L956](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L956): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.269 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L667](https://github.com/Minecolonies/minecolonies/blob/9ad1b4f449c40a8cca3816d9e1240da872741fb0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L667): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 287 issues
* ![MAJOR][MAJOR] 173 major
* ![MINOR][MINOR] 63 minor
* ![INFO][INFO] 51 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutMiner.java#L59](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/client/gui/WindowHutMiner.java#L59): Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L859](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L859): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L956](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L956): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.269 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L667](https://github.com/Minecolonies/minecolonies/blob/a043523248faaf6c5c2607a4af4bc8d27b09637f/src/main/java/com/minecolonies/coremod/colony/Colony.java#L667): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: seems alright

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1158/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by marchermans** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1158/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**